src/run_spineopt_sp.jl

```diff
@@ -383,6 +383,19 @@ function save_outputs!(m)
             new_k = _drop_key(k, :t)
             push!(get!(existing, new_k, Dict{DateTime,Any}()), start(k.t) => v)
         end
+        copy_existing = copy(existing)
```

Comment:
Sorry for insisting, but I'm still trying to understand this:
- why do we need to `copy_existing = copy(existing)` if we will overwrite `copy_existing[k]` in every iteration of the loop?
- why do we need `copy_existing[k] = Dict()` in every loop? It will suffice to have a local variable `d = Dict()`, since we never use `copy_existing` outside the loop.
- `t_e_s` is an array of time slices that are withing each `t_out`, but I don't understand the abbreviation.
- Here: `output_time_slice(m, temporal_block = o)`, `o` is an object of class `output`, why do we pass it to the `temporal_block` argument?
- Is this the right place to do the output aggregation? This part of the loop is skipped if the output name is recognized as a variable or input parameter.

Do the above questions make sense? What am I missing @mihlema ?

Reply:
I think I understand what's going on here, it was just a little bit too clever for me :)

Will try to refactor it for readibility if you don't mind...

Reply:
Sorry I know you reminded me of this but got lost again. But maybe it seems you've already answered yourself? I also noticed that the functionality is compromised in the master (apparently my minimal testing of the output aggregation was not sufficient). Let me know if you need help refactoring, otherwise I can also take this up